### PR TITLE
Add Vibrato

### PR DESCRIPTION
**Pull request by kampersanda**
This PR added a new third-party library, Vibrato. 

Vibrato is evaluated with the following three corpora:

 - `ipadic-mecab v2.7.0`
 - `unidic-mecab v2.1.2`
 - `unidic-cwj v3.1.0`

The system dictionaries are prepared through scripts provided by `thirdparty/vibrato`.
Thus, to keep consistency, I've specified the cargo's dependency with the relative path to `thirdparty/vibrato`.

```
[dependencies]
vibrato = { path = "../../thirdparty/vibrato/vibrato", default-features = false, features = ["unchecked"] }
```

Note that `features = ["unchecked"]` allows for faster tokenization.

### CODE REVIEW

**Comment by kampersanda** [comment]
I hope "Squash and merge" because this PR contains many unnecessary commit logs.

File: bench/vibrato-bench/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+use std::env;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+
+use vibrato::{Dictionary, Tokenizer};
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
```
Comment (vbkaisetsu) [inline bench/vibrato-bench/src/main.rs]: Could you use clap instead of parsing arguments manually?

File: bench/vibrato-bench/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    let rootdir = env!("CARGO_MANIFEST_DIR");
+    let dictname = &args[1];
+
+    let reader = BufReader::new(
+        File::open(format!("{}/resources_{}/system.dic", rootdir, dictname)).unwrap(),
```
Comment (vbkaisetsu) [inline bench/vibrato-bench/src/main.rs]:
```suggestion
        File::open(format!("{rootdir}/resources_{dictname}/system.dic")).unwrap(),
```

File: bench/vibrato-bench/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    );
+    let dict = unsafe { Dictionary::read_unchecked(reader).unwrap() };
+    let mut tokenizer = Tokenizer::new(&dict);
```
Comment (vbkaisetsu) [inline bench/vibrato-bench/src/main.rs:24]: Not essential to this PR issue, but what is the reason Tokenizer is mutable?

If mutable is required, it becomes practical disadvantages, such as the inability to use the tokenizer from multiple threads when building a search index.

File: stats.py [hunk 4]
```diff
@@ -47,6 +50,8 @@ def _main():
                 break
 
     for name, _ in RE_DICT:
+        if not name in times.keys():
+            continue
```
Comment (vbkaisetsu) [inline stats.py]: I think this is strange.

If the `name` is not contained in `times`, that means the script has a bug.

File: stats.py [hunk 5]
```diff
@@ -47,6 +50,8 @@ def _main():
                 break
 
     for name, _ in RE_DICT:
+        if not name in times.keys():
```
Comment (vbkaisetsu) [inline stats.py]:
```suggestion
        if name not in times:
```

File: bench/vibrato-bench/src/main.rs (hunk 1 above)
Comment (kampersanda) [reply bench/vibrato-bench/src/main.rs]: https://github.com/legalforce-research/tokenizer-speed-bench/pull/2/commits/bc813a5df89be0697b84b5341d5b88597748442f

File: stats.py (hunk 4 above)
Comment (kampersanda) [reply stats.py]: The absence of this IF condition forces the results to be parsed to include all libraries defined in `run_all.sh`.
This was inconvenient when testing a part of libraries, so I modified it. But, on second thought, a manual comment out would be sufficient. I reverted it.

Comment (kampersanda) [reply stats.py]: https://github.com/legalforce-research/tokenizer-speed-bench/pull/2/commits/67131af40a501cb8b5fbfb06ad79ef77de2d197f

File: bench/vibrato-bench/src/main.rs (hunk 3 above)
Comment (kampersanda) [reply bench/vibrato-bench/src/main.rs:24]: `Tokenizer` reuses once-allocated memory for quickly constructing a lattice graph, resulting in requiring `mut`. The current version is designed for single-threaded use. Of course, extension to multi-threading would be an important future task.

**Review by vbkaisetsu — APPROVED** [review]

**Pull request merged**